vendor/go-micro.dev/v4/api/options.go

```diff
@@ -16,10 +19,19 @@ func NewOptions(opts ...Option) Options {
 	return options
 }
 
-// WithRouter sets the router to use e.g static or registry
+// WithRouter sets the router to use e.g static or registry.
 func WithRouter(r router.Router) Option {
 	return func(o *Options) error {
 		o.Router = r
 		return nil
 	}
 }
+
+// WithRegistry sets the api's client and router to use registry.
+func WithRegistry(r registry.Registry) Option {
```

Comment:
@butonic Does that change break anything how we use the go micro client?

Reply:
WithRegistry is optional. We always set the registry in out ocis-pkg http and grpc client and server packages.